app/src/main/java/app/nameplaceholder/sensorsapp/MainActivity.kt

```diff
@@ -32,8 +32,6 @@ class MainActivity : AppCompatActivity(), CoroutineScope {
                                 .joinToString(separator = "\n") { sensorData -> sensorData.getPrintable() }
                         }
                     }
-                } else {
-                    App.instance.repository.stop()
```

Comment:
This changes app behavior

Reply:
How?
The stop will be called every time the service stops. Which will happen in the button click.